.github/workflows/ci-cd.yml

```diff
@@ -1,35 +1,52 @@
-name: Tests
-on: [merge_group, push, pull_request, workflow_dispatch]
+name: CI/CD
+on:
+  - merge_group
+  - pull_request
+  - push
+  - workflow_dispatch
 
 env:
   FORCE_COLOR: 1
 
 jobs:
   pre-commit:
-    name: Run Quality Assurance
+    name: Run pre-commit
     runs-on: ubuntu-latest
     steps:
-      - name: Check Out Repository
+      - name: Check out Repository
         uses: actions/checkout@v4
-      - name: Set Up Python
+
+      - name: Set up Python
         uses: actions/setup-python@v4.7.1
         with:
           python-version: '3.11'
+
       - name: Run pre-commit
         uses: pre-commit/action@v3.0.0
 
   test:
-    name: 'Test: Python=${{ matrix.python-version }} on ${{ matrix.os }}'
+    name: 'Test: Python ${{ matrix.python-version }} on ${{ matrix.os }}'
     runs-on: ${{ matrix.os }}
-    needs: [pre-commit]
+    needs:
+      - pre-commit
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest, macOS-latest]
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12', 'pypy-3.8']
-
+        os:
+          - macOS-latest
+          - ubuntu-latest
+          - windows-latest
+        python-version:
+          - '3.8'
+          - '3.9'
+          - '3.10'
+          - '3.11'
+          - '3.12'
+          - 'pypy-3.9'
```

Comment:
`pypy-3.8` removed?

Reply:
Whereas this one was intentional -- I looked at https://www.pypy.org/download.html and it seems 3.8 is not in the latest package version. I don't mind keeping it though if anyone suggests it's worth it.